ray-operator/controllers/ray/utils/dashboard_httpclient.go

```diff
@@ -333,7 +326,14 @@ func (r *RayDashboardClient) ConvertServeConfig(specs []rayv1alpha1.ServeConfigS
 		}
 	}
 
-	return serveConfigToSend
+	servingClusterDeployments := ServingClusterDeployments{
+		ImportPath:  configV1Spec.ImportPath,
+		RuntimeEnv:  applicationRuntimeEnv,
+		Deployments: convertedDeploymentSpecs,
+		Port:        configV1Spec.Port,
```

Comment:
Why do we add `Port` here? The behavior seems to be different from the following code snippet.

https://github.com/ray-project/kuberay/blob/0ae9fc196a50549710b4c4ef8833e3f9e1025f55/ray-operator/controllers/ray/rayservice_controller.go#L662-L666

Reply:
It's part of the spec. The lines you linked here were only used for printing out the json in logs, I think it just missed the port. In `UpdateDeployments` port was included before as well.

Reply:
I see. I believe this is [existing functionality](https://github.com/zcin/kuberay/blob/master/ray-operator/controllers/ray/utils/dashboard_httpclient.go/#L173-L178) so I don't want to change it in this PR - can you create a separate issue for this?

Reply:
I am fine with this change, but the corresponding function for this code snippet appears to be [this](https://github.com/ray-project/kuberay/blob/0ae9fc196a50549710b4c4ef8833e3f9e1025f55/ray-operator/controllers/ray/rayservice_controller.go#L662-L666) instead of `UpdateDeployments` in dashboard_httpclient.go.

Reply:
Yeah, if you look closely that code snippet is only used for print a json to the logs. It's not actually sent to the Serve API.

Reply:
I will open an issue later and resolve this comment. However, it seems that the ports for containers in a Pod cannot be updated dynamically. See [this link](https://stackoverflow.com/questions/57670263/how-to-change-the-port-of-a-kubernetes-container-pod#:~:text=The%20port%20information%20is%20inside,other%20way%20to%20do%20it.) for more details. In this case, the port can only be updated if users expose multiple ports, of which only a few will be used. I am not familiar with serving, so I'm unsure if there are any use cases for this situation.